engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/ProcessCompiler.scala

```diff
@@ -96,8 +97,13 @@ protected trait ProcessCompilerBase {
 
   private val expressionCompiler = ExpressionCompiler.withoutOptimization(classLoader, expressionConfig)
 
-  private def contextWithOnlyGlobalVariables = ValidationContext(Map.empty,
-    expressionConfig.globalVariables.mapValuesNow(_.returnType) + (Interpreter.MetaParamName -> Typed[MetaVariables]))
+  private def contextWithOnlyGlobalVariables(implicit metaData: MetaData): ValidationContext = {
+    val globalTypes = expressionConfig.globalVariables.mapValuesNow(_.returnType)
+    val typesWithMeta = MetaVariables.withType(globalTypes)
+    val typesWithProperties = PropertiesVariable.withType(typesWithMeta)
```

Comment:
IMO would be better to have only one variable meta with nested properties (maybe extracted?). It is simple to say to user, that he/she has in context: global variables, variables created in process and "meta" property and he/she will check what is available in the variable by typing. If you put more than one additional variable to context, than you will need to maintain documentation for all available variables in context.

Reply:
good point, I've nested `properties` under `meta`